<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Storing Data (Need) Exercises</span></div>

# Exercise 1 (Total recall?) #

1. The two similarities between lists and arrays are (1) both are able to store data in an order, and (2) elements in both lists and arrays can be assessed through indexing.
1. The two differences between lists and arrays are (1) lists are in python environment while arrays are in the numpy package, and (2) lists will not be able to get summary statistics such as sum (i.e. py_list.sum will not run) while arrays will be able to do so (i.e. np.sum will work).
1. A dictionary is able to hold data paired with a key, it would be similar to storing 2 lists of data and associating each list's data with each other.

# Exercise 2 (Indexing) #

In [1]:
py_list = ["a1", "b2", "c3", "d4", "e5", "f6", "g7", "h8", "i9", "j10"]
print(py_list[0])    # Prints 'a1'
print(py_list[2])    # Prints 'c3'
print(py_list[4])    # Prints 'e5'
print(py_list[6])    # Prints 'g7'
print(py_list[8])    # Prints 'i9'

a1
c3
e5
g7
i9


# Exercise 3 (Index again) #

In [2]:
elements = ['Hydrogen',
            'Helium', 'Lithium',
            'Beryllium', 'Boron', 'Carbon',
            'Nitrogen', 'Oxygen',
            'Fluorine',
            'Neon']

print(elements[4])
print(elements[-6])

Boron
Boron


# Exercise 4 (How many ones) #

In [3]:
import numpy as np

In [5]:
numbers=[45, 60, 1, 30, 96, 1, 96, 57, 16, 1,
        99, 62, 86, 43, 42, 60, 59, 1, 1, 35,
        83, 47, 34, 28, 68, 23, 22, 92, 1, 79,
        1, 29, 94, 72, 46, 47, 1, 74, 32, 20,
        8, 37, 35, 1, 89, 29, 86, 19, 43, 61]                  

np_numbers = np.array(numbers)

sum(np_numbers == 1)

np.int64(9)

There are 9 "1"s in the list.

# Exercise 5 (A Matter of Statistics) #

In [5]:
import numpy as np

In [8]:
x_1 = [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
x_2 = [1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

np_x1 = np.array(x_1)
np_x2 = np.array(x_2)

In [9]:
np.mean(np_x1)

np.float64(38.5)

In [10]:
np.mean(np_x2)

np.float64(302.5)

The mean of x<sub>1</sub> is 38.5, while the mean of x<sub>2</sub> is 302.5.

In [11]:
np.std(np_x1)

np.float64(32.41990129534635)

In [13]:
np.std(np_x2)

np.float64(326.0893282522444)

The standard deviation of x<sub>1</sub> to 3 d.p. is 32.420, while the standard deviation of x<sub>2</sub> to 3 d.p. is 326.089.

In [14]:
((10 * np.sum(np_x1 * np_x2) - (np.sum(np_x1) * np.sum(np_x2)))) / np.sqrt((10 * np.sum(np_x1 ** 2) - (np.sum(np_x1) ** 2)) * (10 * np.sum(np_x2 ** 2) - (np.sum(np_x2) ** 2)))

np.float64(0.9871796626270701)

In [24]:
np.corrcoef(np_x1, np_x2)

array([[1.        , 0.98717966],
       [0.98717966, 1.        ]])

Both codes gave the same result for the Pearson correlation coefficient _r_ of 0.98717966.